backend/server/routers/programs.py

```diff
@@ -158,11 +160,26 @@ def convertSubgroupObjectToCoursesDict(object: str, description: str|list[str]):
 
 def addSubgroupContainer(structure: dict, type: str, container: dict, exceptions: list[str]) -> list[str]:
     """ Returns the added courses """
-    structure[type][container["title"]] = {}
-    item = structure[type][container["title"]]
-
+    #TODO: further standardise non_spec_data to remove these line:
+    title = container.get("title")
+    if container.get("type") == "gened":
+        title = "General Education"
+    
+    # don't do anything if it's info_rule or limit_rule
+    if container.get("type") is not None and "rule" in container.get("type"):
+        return []
+
+    if container.get("credits_to_complete") is None:
```

Comment:
It's impossible for it to be None anymore, it'll be set to 0 if there's some error

Reply:
(can change if wanted)

Reply:
yup, changed to 0 by default